File: frontend/app/src/main/java/com/smartcity/app/DisplayLabels.java
package com.smartcity.app;

import android.content.Context;

/**
 * Traduit les valeurs API (anglais) en libellés français pour l'affichage uniquement.
 * Les clés JSON (safety, pending, etc.) ne sont jamais modifiées.
 */
public final class DisplayLabels {

    private DisplayLabels() {}

    public static String category(Context context, String apiCategory) {
        if (apiCategory == null || apiCategory.isEmpty()) {
            return context.getString(R.string.display_cat_general);
        }
        switch (apiCategory.toLowerCase()) {
            case "safety":
                return context.getString(R.string.display_cat_safety);
            case "infrastructure":
                return context.getString(R.string.display_cat_infrastructure);
            case "lighting":
                return context.getString(R.string.display_cat_lighting);
            case "waste":
                return context.getString(R.string.display_cat_waste);
            case "water":
                return context.getString(R.string.display_cat_water);
            default:
                return context.getString(R.string.display_cat_general);
        }
    }

    public static String status(Context context, String apiStatus) {
        if (apiStatus == null || apiStatus.isEmpty()) {
            return context.getString(R.string.display_stat_pending);
        }
        switch (apiStatus.toLowerCase()) {
            case "resolved":
                return context.getString(R.string.display_stat_resolved);
            case "in_progress":
                return context.getString(R.string.display_stat_in_progress);
            case "pending":
            default:
                return context.getString(R.string.display_stat_pending);
        }
    }
}

File: backend/app/Support/DisplayLabels.php
<?php

namespace App\Support;

/**
 * Libellés français pour l'affichage des valeurs API (catégories IA, statuts).
 * Les clés en base / JSON restent en anglais.
 */
final class DisplayLabels
{
    private const CATEGORIES = [
        'safety' => 'Sécurité',
        'infrastructure' => 'Infrastructure',
        'lighting' => 'Éclairage',
        'waste' => 'Déchets',
        'water' => 'Eau',
        'general' => 'Général',
    ];

    private const STATUSES = [
        'pending' => 'En attente',
        'in_progress' => 'En cours',
        'resolved' => 'Résolu',
    ];

    private function __construct() {}

    public static function category(?string $apiCategory): string
    {
        if ($apiCategory === null || $apiCategory === '') {
            return self::CATEGORIES['general'];
        }

        return self::CATEGORIES[strtolower($apiCategory)] ?? self::CATEGORIES['general'];
    }

    public static function status(?string $apiStatus): string
    {
        if ($apiStatus === null || $apiStatus === '') {
            return self::STATUSES['pending'];
        }

        return self::STATUSES[strtolower($apiStatus)] ?? self::STATUSES['pending'];
    }
}
